### PR TITLE
removing more indexes

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

File: corehq/apps/accounting/migrations/0041_remove_like_indexes.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+        db.execute("DROP INDEX IF EXISTS accounting_defaultproductplan_product_type_like")
+        db.execute("DROP INDEX IF EXISTS accounting_feature_feature_type_like")
+        # db.execute("DROP INDEX IF EXISTS accounting_feature_name_like")
+        db.execute("DROP INDEX IF EXISTS accounting_paymentmethod_method_type_like")
+        # db.execute("DROP INDEX IF EXISTS accounting_softwareplan_name_like")
+        # db.execute("DROP INDEX IF EXISTS accounting_softwareproduct_name_like")
+        db.execute("DROP INDEX IF EXISTS accounting_softwareproduct_product_type_like")
+        db.execute("DROP INDEX IF EXISTS accounting_subscriber_domain_like")
+        # db.execute("DROP INDEX IF EXISTS accounting_subscriber_organization_like")
+        # db.execute("DROP INDEX IF EXISTS accounting_wirebillingrecord_emailed_to_like")
```
Comment (snopoke) [inline corehq/apps/accounting/migrations/0041_remove_like_indexes.py]: @biyeun @nickpell please review

The only 'like' queries I could find in the code were on `BillingAccount.name` and `SoftwarePlan.name`. Can I remove any of the other commented out ones?

File: corehq/apps/smsbillables/migrations/0011_remove_like_indexes.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    def forwards(self, orm):
+        db.execute("DROP INDEX IF EXISTS smsbillables_smsbillable_direction_like")
+        db.execute("DROP INDEX IF EXISTS smsbillables_smsbillable_domain_like")
+        db.execute("DROP INDEX IF EXISTS smsbillables_smsgatewayfeecriteria_backend_api_id_like")
+        db.execute("DROP INDEX IF EXISTS smsbillables_smsgatewayfeecriteria_backend_instance_like")
+        db.execute("DROP INDEX IF EXISTS smsbillables_smsgatewayfeecriteria_direction_like")
+        db.execute("DROP INDEX IF EXISTS smsbillables_smsgatewayfeecriteria_prefix_like")
+        db.execute("DROP INDEX IF EXISTS smsbillables_smsusagefeecriteria_direction_like")
+        db.execute("DROP INDEX IF EXISTS smsbillables_smsusagefeecriteria_domain_like")
```
Comment (snopoke) [inline corehq/apps/smsbillables/migrations/0011_remove_like_indexes.py]: @gcapalbo please review

File: custom/ilsgateway/migrations/0012_remove_like_indexes.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    def forwards(self, orm):
+        db.execute("DROP INDEX IF EXISTS ilsgateway_alert_supply_point_like")
+        # db.execute("DROP INDEX IF EXISTS ilsgateway_deliverygroupreport_message_like")
+        db.execute("DROP INDEX IF EXISTS ilsgateway_deliverygroupreport_supply_point_like")
+        db.execute("DROP INDEX IF EXISTS ilsgateway_organizationsummary_supply_point_like")
+        db.execute("DROP INDEX IF EXISTS ilsgateway_productavailabilitydata_product_like")
+        db.execute("DROP INDEX IF EXISTS ilsgateway_productavailabilitydata_supply_point_like")
+        db.execute("DROP INDEX IF EXISTS ilsgateway_requisitionreport_location_id_like")
+        db.execute("DROP INDEX IF EXISTS ilsgateway_supplypointstatus_supply_point_like")
+        db.execute("DROP INDEX IF EXISTS ilsgateway_supplypointwarehouserecord_supply_point_like")
```
Comment (snopoke) [inline custom/ilsgateway/migrations/0012_remove_like_indexes.py]: @czue please review

File: corehq/apps/smsbillables/migrations/0011_remove_like_indexes.py (hunk 2 above)
Comment (gcapalbo) [inline corehq/apps/smsbillables/migrations/0011_remove_like_indexes.py]: will defer to @nickpell on the smsbillables ones, but I'm good with dropping the like indexes on the sms_* and smsforms_* tables

**Comment by dannyroberts** [comment]
Weird, so Django by default adds "like" indexes to anything you decide to index—and doesn't give you a way to specify anything other than that default in your ORM fields/models, other than manually deleting them with raw SQL in migrations!? Nice find...

File: corehq/apps/smsbillables/migrations/0011_remove_like_indexes.py (hunk 2 above)
Comment (nickpell) [inline corehq/apps/smsbillables/migrations/0011_remove_like_indexes.py]: :+1:

**Comment by snopoke** [comment]
Yup, but only for text columns and only when it initially creates the
column. If you add the index after the column is created then you don't get
the 'like' index.

**Comment by dannyroberts** [comment]
lol—"i mean obviously"

File: corehq/apps/accounting/migrations/0041_remove_like_indexes.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/accounting/migrations/0041_remove_like_indexes.py]: I would keep ```accounting_billingaccount_salesforce_account_id_like```

You can remove ```accounting_subscriber_organization_like``` and ```accounting_wirebillingrecord_emailed_to_like``` (@benrudolph)

Comment (benrudolph) [inline corehq/apps/accounting/migrations/0041_remove_like_indexes.py]: yep

**Comment by benrudolph** [comment]
that's so absurd

File: custom/ilsgateway/migrations/0012_remove_like_indexes.py (hunk 3 above)
Comment (czue) [inline custom/ilsgateway/migrations/0012_remove_like_indexes.py]: :+1: cc @kkrampa but I'm pretty sure we don't do any like queries anywhere in the ils/ews code

**Comment by czue** [comment]
yeah this is great and terrible.

**Comment by czue** [comment]
I think everyone has signed off. @snopoke feel free to self-merge if you agree.

**Comment by czue** [comment]
er, and after billing stuff has been updated

**Comment by snopoke** [comment]
Merged this. Also tracked down the root of the extra indexes: https://code.djangoproject.com/ticket/12234

**Pull request merged**